### DISCUSSION TITLE
homework a fortune teller program

### DISCUSSION THREAD

**Discussion by twlee2** [cpp-for-yourself/lectures-and-homeworks#22] [Homeworks Q&A] [answered]
It seems we need to print "Here is your description:" right before printing the fortune to get green checks while testing.

**Comment by niosus** [comment, marked as answer]
Oh my, sorry, I did not see this comment here. :facepalm: 

I will have a look tonight and get back to you on this.

Just to make sure I understand you correctly, what exactly is your question here?

**Reply by twlee2** [reply]
No worries :) Thank you for looking at this issue.
My question is: Does homework bot require us to cout "Here is your description:" right before printing the fortune telling?
For example,
**Here is your description:**
Igor, the fearless STL guru that finds errors quicker than the compiler
In my case, it seems to require the additional line to get green checks

**Reply by niosus** [reply]
Well, it should be exactly like it is in the video. Also, the homework bot tells you what it expects to see, right? Am I missing the point here?

**Reply by twlee2** [reply]
right, you are not missing anything here. So, a takeaway here is that the homework bot will be happy when it sees **exactly** what it expects to see. Thank you!

**Reply by niosus** [reply]
Yeah, it should be, just like it is shown [in the video](https://youtu.be/GCh8VDZRV4k?t=41). I will add this to the homework description too!

**Reply by ilarioazzollini** [reply]
Hi @niosus, I got the same issue

```
--- Actual output

+++ Expected output

@@ -4,4 +4,6 @@

 (pick from 'spring', 'summer', 'autumn', 'winter')
 Please enter an adjective:
 Please enter another adjective:
+
+Here is your description:
 You, the clever STL guru that eats UB for breakfast
```
 
This is obviously a minor issue and the bot message clearly tells you what it needs to be happy.

Anyway, what is misleading in my opinion is that in the [instructions](https://github.com/cpp-for-yourself/supplementary-materials/blob/main/homeworks/homework_2/homework.md) the "Here is your description:" string is never mentioned.

Thank you for your work and support! Cheers!

**Reply by compi2k** [reply]
why was i mentioned?

**Reply by jbouze** [reply]
> Hi @niosus, I got the same issue
> 
> ```
> --- Actual output
> 
> +++ Expected output
> 
> @@ -4,4 +4,6 @@
> 
>  (pick from 'spring', 'summer', 'autumn', 'winter')
>  Please enter an adjective:
>  Please enter another adjective:
> +
> +Here is your description:
>  You, the clever STL guru that eats UB for breakfast
> ```
> 
> This is obviously a minor issue and the bot message clearly tells you what it needs to be happy.
> 
> Anyway, what is misleading in my opinion is that in the [instructions](https://github.com/cpp-for-yourself/supplementary-materials/blob/main/homeworks/homework_2/homework.md) the "Here is your description:" string is never mentioned.
> 
> Thank you for your work and support! Cheers!

This stumped me as well for a while - just submitted a pull request to correct the homework_2 instructions as I won't be the last.

**Reply by niosus** [reply]
Thanks a lot for pointing this out and for creating a PR too @jbouze, let's continue our discussion there and merge it.
